compiler/rustc_errors/src/lib.rs

```diff
@@ -552,6 +561,11 @@ struct DiagCtxtInner {
     /// The delayed bugs and their error guarantees.
     delayed_bugs: Vec<(DelayedDiagInner, ErrorGuaranteed)>,
 
+    /// *specifically* for lints emitted during ast lowering.
+    /// At this point we can't get the lint level yet (doing so will cause really weird bugs).
+    /// So, by stuffing lints in here they can be emitted when hir is built.
+    hir_delayed_lints: Vec<HirDelayedLint>,
```

Comment:
I don't think this is sound (wrt incremental) for lints. I think the right place would be to put them into https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/struct.OwnerInfo.html and emit them from the late lint for each owner.

Reply:
should be fixed now